lib/model/rwfolder.go

```diff
@@ -690,20 +692,9 @@ func (f *rwFolder) deleteDir(file protocol.FileInfo, matcher *ignore.Matcher) {
 		f.newError(file.Name, err)
 		return
 	}
-	// Delete any temporary files lying around in the directory
-	dir, _ := os.Open(realName)
-	if dir != nil {
-		files, _ := dir.Readdirnames(-1)
-		for _, dirFile := range files {
-			fullDirFile := filepath.Join(file.Name, dirFile)
-			if defTempNamer.IsTemporary(dirFile) || (matcher != nil && matcher.Match(fullDirFile).IsDeletable()) {
-				os.RemoveAll(filepath.Join(f.dir, fullDirFile))
-			}
-		}
-		dir.Close()
-	}
 
-	err = osutil.InWritableDir(os.Remove, realName)
+	err = deletedir(f.dir, file, matcher)
```

Comment:
The fact that this logic has moved from rwfolder to some function that is not part of the struct, that is now used by the model smells really bad.
As I said, I think you should make a:

```
type rofolder struct {
  rwfolder
}
```

and override the few places that are relevant to you.

We print folder type somewhere in model.StartFolder() which will definately misreport in your case.